packages/beacon-node/src/node/nodejs.ts

```diff
@@ -187,6 +185,27 @@ export class BeaconNode {
       signal.addEventListener("abort", metrics.close, {once: true});
     }
 
+    const clock = new Clock({config, genesisTime: anchorState.genesisTime, signal});
+
+    // Prune hot db repos
+    // TODO: Should this call be awaited?
+    await db.pruneHotDb();
+
+    if (opts.chain.pruneHistory) {
+      // prune ALL stale data before starting
+      logger.info("Pruning historical data");
```

Comment:
do we know how fast this is if the db is huge? I can test this on my node but might wanna add some period logs so that the user can some feedback on what's going on, similar to

https://github.com/ChainSafe/lodestar/blob/9c66fac4a47446b225c874997dfc4d7b93a820b3/packages/cli/src/networks/index.ts#L180